### PR TITLE
Add vlanId validation and improve the old validation, update related unit-test.

### PR DESCRIPTION
**Pull request by changev**
1. Add vlanId and netmask validation, vlanId must between 0 - 4095, netmask must be legal.
2. Improve the ipAddress/gateway  validation by Assert module.
3. Update related unit-test by add a new descirbe.
@RackHD/corecommitters @yyscamper

### CODE REVIEW

**Comment by changev** [comment]
jenkins test this please

**Comment by yyscamper** [comment]
@jlongever , do you know why the jenkins refuses to test this PR?

File: lib/jobs/install-os.js [hunk 1]
```diff
@@ -83,16 +83,54 @@ function installOsJobFactory(
...
+                        //judge if a number is like '11110000' or '0' which a netmask is.
+                        /* jshint ignore:start */
+                        if (item !== 0 && (item - 1 | item) !== 255) {
+                            throw new Error('Invalid ipv4 netmask.');
+                        }
+                        /* jshint ignore:end */
+                    });
+
+                    vlanIds = dev.ipv4.vlanId;
+                    assert.array(vlanIds);
```
Comment (yyscamper) [inline lib/jobs/install-os.js]: the vlanId is just optional, so you should validate it only if user has some input for it.

**Comment by jlongever** [comment]
Because @changev is not a white listed user.

Jenkins,  ok to test

**Comment by panpan0000** [comment]
:+1:

**Comment by yyscamper** [comment]
jenkins: add to whitelist

**Comment by changev** [comment]
jenkins test this please

**Comment by yyscamper** [comment]
:+1:

**Pull request merged**